### PR TITLE
feat(inspector): Allow including/excluding safelist in REPL

### PR DESCRIPTION
**Pull request by jd-solanki**
Closes: #1530

Excluding safelist:
Screenshot 2022-09-03 at 9 51 50 AM

Including safelist:
Screenshot 2022-09-03 at 9 52 07 AM

---

_P.S. There are no recommended extensions listed in `.vscode`. Moreover, if there's ESLint I suggest adding your eslint config as well._

### CODE REVIEW

File: packages/inspector/client/components/ReplPlayground.vue [hunk 1]
```diff
@@ -10,7 +10,8 @@ const input = useLocalStorage(
   '<div class="text-sm hover:text-red">\nHello World\n</div>',
 )
 
-const { data: result } = fetchRepl(input)
+const isSafelistIncluded = ref(false)
```
Comment (antfu) [inline packages/inspector/client/components/ReplPlayground.vue:13]:
```suggestion
const isSafelistIncluded = useStorage('unocss-inspector-safelist', false)
```

File: packages/inspector/node/index.ts [hunk 2]
```diff
@@ -65,8 +65,9 @@ export default function UnocssInspector(ctx: UnocssPluginContext): Plugin {
       if (req.url.startsWith('/repl')) {
         const query = new URLSearchParams(req.url.slice(5))
         const token = query.get('token') || ''
+        const includeSafelist = JSON.parse(query.get('include-safelist') ?? 'false')
```
Comment (antfu) [inline packages/inspector/node/index.ts:68]:
```suggestion
        const includeSafelist = JSON.parse(query.get('safelist') ?? 'false')
```

File: packages/inspector/client/composables/fetch.ts [hunk 3]
```diff
@@ -32,9 +32,9 @@ export function fetchModule(id: string | Ref<string>) {
   return result
 }
 
-export function fetchRepl(input: Ref<string>) {
+export function fetchRepl(input: Ref<string>, includeSafelist: Ref<boolean>) {
   const debounced = useDebounce(input, 500)
-  return useFetch(computed(() => `${API_ROOT}/repl?token=${encodeURIComponent(debounced.value)}`), { refetch: true })
+  return useFetch(computed(() => `${API_ROOT}/repl?token=${encodeURIComponent(debounced.value)}&include-safelist=${includeSafelist.value}`), { refetch: true })
```
Comment (antfu) [inline packages/inspector/client/composables/fetch.ts:37]:
```suggestion
  return useFetch(computed(() => `${API_ROOT}/repl?token=${encodeURIComponent(debounced.value)}&safelist=${includeSafelist.value}`), { refetch: true })
```

**Comment by jd-solanki** [comment]
Thanks for the suggestions!

Done 👍🏻

**Comment by antfu** [comment]
Can you help testing if the new changes works for you?

**Comment by jd-solanki** [comment]
> testing if the new changes work for you

I linked unocss to fresh vuejs project and used this config:
```js
Unocss({
    safelist: ['bg-red', 'opacity-50'],
    blocklist: ['bg-red'],
})
```

REPL:
Screenshot 2022-09-03 at 12 43 33 PM
Screenshot 2022-09-03 at 12 43 48 PM

**Yes, It's working**

**Comment by antfu** [comment]
Thanks

**Pull request merged**